### PR TITLE
MINOR: Add system configuration to zk security exception messages

### PR DESCRIPTION
**Pull request by hachikuji**
This patch ensures that relevant system configurations are included in exception messages when zk security validation fails.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by hachikuji** [comment]
@rajinisivaram @omkreddy Does this seem reasonable?

File: clients/src/main/java/org/apache/kafka/common/security/JaasUtils.java [hunk 1]
```diff
@@ -31,27 +31,48 @@
...
 
+    private static final String DEFAULT_ZK_LOGIN_CONTEXT_NAME = "Client";
+    private static final String DEFAULT_ZK_SASL_CLIENT = "true";
+
     private JaasUtils() {}
 
+    public static String zkSecuritySysConfigString() {
+        String loginConfig = System.getProperty(JAVA_LOGIN_CONFIG_PARAM);
+        String clientEnabled = System.getProperty(ZK_SASL_CLIENT);
+        String contextName = System.getProperty(ZK_LOGIN_CONTEXT_NAME_KEY);
```
Comment (jsancio) [inline clients/src/main/java/org/apache/kafka/common/security/JaasUtils.java]: How about the following to simplify the string construction below:
```java
        String clientEnabled = System.getProperty(ZK_SASL_CLIENT, "default:" + DEFAULT_ZK_SASL_CLIENT);
        String contextName = System.getProperty(ZK_LOGIN_CONTEXT_NAME_KEY, "default:" + DEFAULT_ZK_LOGIN_CONTEXT_NAME);
```

**Review by jsancio — APPROVED** [review]
LGTM. One minor comment.

**Review by soondenana — APPROVED** [review]
Nothing to add to Jose's comment. Thanks for improving this, LGTM.

**Review by omkreddy — APPROVED** [review]
@hachikuji Thanks for the PR. LGTM.

**Pull request merged**